spirit/category/admin/views.py

```diff
@@ -55,3 +56,31 @@ def update(request, category_id):
     context = {'form': form, }
 
     return render(request, 'spirit/category/admin/update.html', context)
+
+
+@administrator_required
+def move_up(request, category_id):
+    category = get_object_or_404(Category, pk=category_id)
+    sibling = Category.objects.filter(parent=category.parent, sort__lt=category.sort)\
+        .exclude(pk=settings.ST_TOPIC_PRIVATE_CATEGORY_PK).order_by('-sort')[::1]
```

Comment:
I'm assuming this should be `[:1]`, but use `first()` instead. Then you can remove all the random access, i.e just `sibling` instead of `sibling[0]`